force-app/main/default/classes/AccountAutoDeletionSettingsVMapper.cls

```diff
@@ -0,0 +1,124 @@
+     **********************************************************************************/
+    protected AccountAutoDeletionSettingsVMapper() {
+    }
+
+    /**********************************************************************************
+     * @description Static method to get the current instance for Singleton pattern
+     * @return The instance of AccountAutoDeletionSettingsVMapper.
+     **********************************************************************************/
+    public static AccountAutoDeletionSettingsVMapper getInstance() {
+        if (instance == null) {
+            instance = new AccountAutoDeletionSettingsVMapper();
+        }
+
+        return instance;
+    }
+
+    /*******************************************************************************************************
+     * @description Retrieve a DualListboxVModel representing the Account Record Types currently selected
+     * for deletion when not associated with Contact records and any remaining active account record types
+     * availabe for selection.
+     * @return  A DualListboxVModel
+     *******************************************************************************************************/
+    public virtual DualListboxVModel getAccountAutoDeletionSettingsViewModel() {
+        // get current values set for deletion from hierarchy settings
+        AccountRecordTypeAutoDeletionModel currentAutoDeletionSettings = this.locateAccountRecordTypeAutoDeletionService()
+            .getAccountRecordTypeAutoDeletionModel();
+
+        // get all active account record types in org
+        Map<Id, RecordTypeModel> activeAccountRecordTypeModelsById = this.locateRecordTypeService()
+            .getActiveRecordTypeModelsById(
+                this.locateRecordTypeService().getModelByIdMapForSObjectType(Account.SObjectType).values()
+            );
+
+        // create DualListboxValueVModels for each of the selected and available options
+        List<String> selectedOptionsDualListboxValuesList = new List<String>();
+        List<DualListboxValueVModel> optionsDualListboxValuesList = new List<DualListboxValueVModel>();
+
+        // populate available listbox - should contain all values regardless of if they are selected
+        for (Id recordTypeId : activeAccountRecordTypeModelsById.keySet()) {
+            optionsDualListboxValuesList.add(
+                new DualListboxValueVModel(
+                    activeAccountRecordTypeModelsById.get(recordTypeId).name,
+                    activeAccountRecordTypeModelsById.get(recordTypeId).id
+                )
+            );
+        }
+
+        // populate selected listbox - only requires the String representing the value of the selected option
+        for (
+            Id accountRecordTypeIdToDelete : (List<Id>) currentAutoDeletionSettings.accountRecordTypeIdsForDeletionWhenOrphanedList
+        ) {
+            if (activeAccountRecordTypeModelsById.containsKey(accountRecordTypeIdToDelete) == false) {
```

Comment:
This logic may be repeatable when we get to addresses and other settings that can get played with. Consider making a filter method into the record type service. Something like filterListForValidIds, which returns a list in the same order and uses a set of record type ids to validate that against. Leaving it this way, for now, is also fine, but that kind of behavior is not unique to this setting.